test/7.compliance.test.js

```diff
@@ -1,124 +1,140 @@
 import {expect} from 'chai';
-import FL from 'fantasy-land';
 import Future from '../index.mjs.js';
-import * as U from './util';
-import jsc from 'jsverify';
+import {assertEqual as eq, I, B, noop, STACKSIZE, error, bang} from './util';
+import {Functor, Bifunctor, Apply, Applicative, Chain, ChainRec, Monad} from 'fantasy-laws';
+import Z from 'sanctuary-type-classes';
+import {
+  bool,
+  constant as _k,
+  falsy,
+  fn,
+  letrec,
+  nat,
+  number,
+  oneof,
+  property,
+  string
+} from 'jsverify';
+
+//  FutureArb :: Arbitrary a -> Arbitrary b -> Arbitrary (Future a b)
+var FutureArb = function(larb, rarb){
+  var toStr = function(m){ return m.toString() };
+  var getLeft = function(m){
+    var x;
+    m.fork(function(y){ x = y }, function(){ throw error });
+    return x;
+  };
+  var getRight = function(m){
+    var x;
+    m.fork(function(){ throw error }, function(y){ x = y });
+    return x;
+  };
+  return oneof(
+    larb.smap(Future.reject, getLeft, toStr),
+    rarb.smap(Future.of, getRight, toStr),
+    larb.smap(function(x){ return Future.reject(x).map(I) }, getLeft, toStr),
+    rarb.smap(function(x){ return Future.of(x).map(I) }, getLeft, toStr)
+  );
+};
+
+//  anyFuture :: Arbitrary (Future Any Any)
+var {anyFuture} = letrec(function(tie){
```

Comment:
This arbitrary is *by far* the ideal one to use in every test, as it allows tests to assert important properties, such as higher order monads, which would otherwise not be asserted.

Unfortunately, I am unable to use it in most of the tests, as there is no way for me to associated arbitrary data with arbitrary functions that operate on said data. If you'll look into the code below, you'll find that functions are always passed as constants.

I have an idea on how to fix this, and possibly make a bug improvement to fantasy-laws in the process, but for the the scope of the solution is too big to be a concern of this PR.